lib/puppet/provider/foreman_resource/rest_v3.rb

```diff
@@ -95,6 +98,21 @@ def success?(response)
   end
 
   def error_message(response)
-    JSON.parse(response.body)['error']['full_messages'].join(' ') rescue "unknown error (response #{response.code})"
+    messages = {
+      '400' => '400 Bad Request: Something is wrong with the data we sent to Foreman server',
+      '401' => '401 Unauthorized Request: Check credentials for validity',
+      '403' => '403 Forbidden: The credentials were valid but do not grant access to the requested resource',
```

Comment:
This is highly unlikely. oauth has full access and there are no permission checks. I don't think we should document this. Just keep it generic:
```suggestion
      '403' => "#{response.code} #{response.message}"
```

Probably even:
```suggestion
      default => "#{response.code} #{response.message}"
```

Reply:
I incorporated this idea outside of `messages` and instead at `JSON.parse(response.body)['error']['full_messages'].join(' ') rescue "Response: #{response.code} #{response.message})"`

Let me know please if that's suitable to you